File: gallery/src/main/java/com/qkopy/gallery/ui/common/BaseRecyclerViewAdapter.java
/*
 * Created by @ajithvgiri on 4/12/18 8:13 PM
 * Copyright (c) 2018 . All rights reserved.
 * Last modified 4/12/18 7:05 PM
 */

package com.qkopy.gallery.ui.common;

import android.content.Context;
import android.view.LayoutInflater;
import androidx.recyclerview.widget.RecyclerView;

public abstract class BaseRecyclerViewAdapter<T extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<T> {
    private final Context context;
    private final LayoutInflater inflater;

    public BaseRecyclerViewAdapter(Context context) {
        this.context = context;
        this.inflater = LayoutInflater.from(context);
    }

    public Context getContext() {
        return context;
    }

    public LayoutInflater getInflater() {
        return inflater;
    }
}

File: gallery/src/main/java/com/qkopy/gallery/ui/common/BaseRecyclerViewAdapter.kt
package com.qkopy.gallery.ui.common

import android.content.Context
import android.view.LayoutInflater
import androidx.recyclerview.widget.RecyclerView

abstract class BaseRecyclerViewAdapter<T : RecyclerView.ViewHolder?>(val context: Context) :
    RecyclerView.Adapter<T>() {
    val inflater: LayoutInflater

    init {
        inflater = LayoutInflater.from(context)
    }
}
